test/powershell/Language/Scripting/NativeExecution/NativeCommandProcessor.Tests.ps1

```diff
@@ -1,22 +1,32 @@
+$testroot = resolve-path (join-path $psscriptroot ../../..)
+$common = join-path $testroot Common
+$helperModule = join-path $common Test.Helpers.psm1
+
 Describe 'native commands lifecycle' -tags 'Feature' {
 
     BeforeAll {
+        Import-Module $helperModule -force
         $powershell = Join-Path -Path $PsHome -ChildPath "powershell"
     }
 
-    #Marking this test as Pending due to issue # https://github.com/PowerShell/PowerShell/issues/2802
-    It "native | ps | native doesn't block" -Pending {
-        $first = $true
-        & $powershell -command '1..10 | % {Start-Sleep -mill 100; $_}' | %{$_} | & $powershell -command '$input' | % {
-            if ($first)
-            {
-                $first = $false
-                $firstTime = [datetime]::Now
-            }
-            $lastTime = [datetime]::Now
-        }
+    It "native | ps | native doesn't block" {
+        $iss = [initialsessionstate]::CreateDefault2();
+        $rs = [runspacefactory]::CreateRunspace($iss)
+        $rs.Name = "TestRunspaceDebuggerReset"
+        $rs.Open()
+
+        $ps = [powershell]::Create()
+        $ps.Runspace = $rs
+
+        # waiting 30 seconds, because powershell startup time could be long on the slow machines,
+        # such as CI
+        Wait-CompleteExecution {
```

Comment:
I don't think `Wait-CompleteExecution` is named accurately - it looks like it evaluates the script block repeatedly until it returns true, not until it's complete, so I'm not sure that's what you actually want.

It also has a tiny bug when the time changes, e.g. going from PDT to PST and back - the date arithmetic does not account for the time change.